bitbots_vision/src/bitbots_vision/vision_modules/obstacle.py

```diff
@@ -28,19 +28,30 @@ def __init__(self, red_color_detector, blue_color_detector, white_color_detector
         self._obstacle_finder_method = config['obstacle_finder_method']
         self._distance_value_increase = config['obstacle_finder_value_increase']
 
+        # Set values to None needed for caching
         self._image = None
         self._blue_mask = None
         self._red_mask = None
         self._white_mask = None
 
+        # Set output to None
         self._obstacles = None
         self._blue_obstacles = None
         self._red_obstacles = None
         self._white_obstacles = None
         self._other_obstacles = None
 
     def set_image(self, image):
+        """
+        Set a image for the obstacle detector. This also resets the caches.
+        :param image: current vision image
+        """
+        # Check if image has been set
+        if np.array_equal(image, self._image):
```

Comment:
You lose about half a millisecond. In this detector, it's necessary because the detector gets called multiple times on the same image and each call sets the image using `set_image`so nothing gets cached. In other detectors where this is not necessary, I would not do it.